src/niftyone/figures/generator.py

```diff
@@ -92,10 +107,27 @@ def __call__(
         overwrite: bool,
     ) -> None:
         # Filters by entity (via string query)
-        table_idxes = table.ent.query(self.query).index
-        for idx in table_idxes:
-            record = table.nested.loc[idx]
-            self.generate(record=record, out_dir=out_dir, overwrite=overwrite)
+        # First query is for main image, subsequent are for overlays
+        query_dfs = [table.ent.query(q) for q in self.queries]
+        indexed_dfs = [
+            (
+                df.assign(table_index=df.index)
+                .reset_index(drop=True)
+                .loc[:, ["table_index"] + self.join_entities]
+                .rename(columns={"table_index": f"index_{ii}"})
+            )
+            for ii, df in enumerate(query_dfs)
+        ]
+        joined = reduce(
+            lambda left, right: pd.merge(
+                left, right, on=self.join_entities, how="outer"
```

Comment:
I'll double check the `how` call. 

The join might work? I can't recall now how I had originally attempted this, but it had thrown an error when more than 1 dataframe was provided.

Reply:
I think you're right re: `inner` vs `outer`, but I'm not sure it matters here since the dfs that it tries to join only contain the values from the join entities and the corresponding index to the query. And since the join is on the entities, they should always match.